src/adjoint_common.jl

```diff
@@ -401,9 +401,15 @@ function (f::ReverseLossCallback)(integrator)
         copyto!(y, integrator.u[(end - idx + 1):end])
     end
 
-    # Warning: alias here! Be careful with λ
-    gᵤ = isq ? λ : @view(λ[1:idx])
-    g(gᵤ, y, p, t[cur_time[]], cur_time[])
+    if ArrayInterfaceCore.ismutable(u)
+        # Warning: alias here! Be careful with λ
+        gᵤ = isq ? λ : @view(λ[1:idx])
+        g(gᵤ, y, p, t[cur_time[]], cur_time[])
+    else
+        @assert sensealg isa QuadratureAdjoint
+        gᵤ = isq ? λ : @view(λ[1:idx])
```

Comment:
```suggestion
        gᵤ = isq ? λ : λ[1:idx]
```

No views are required on immutables.

Reply:
@assert

Reply:
true, even better